graphrag/index/flows/create_final_embeddings.py

```diff
@@ -0,0 +1,211 @@
+    callbacks: VerbCallbacks,
+    cache: PipelineCache,
+    storage: PipelineStorage,
+    embedded_fields: set[str],
+    base_text_embed: dict | None = None,
+) -> pd.DataFrame:
+    """All the steps to generate all embeddings."""
+    documents_embeddings = final_documents[["id", "raw_content"]].copy()
+    relationships_embeddings = final_relationships[["id", "description"]].copy()
+    text_units_embeddings = final_text_units[["id", "text"]].copy()
+
+    entities_embeddings = final_entities[["id", "name", "description"]].copy()
+    entities_embeddings["name_description"] = (
+        entities_embeddings["name"] + ":" + entities_embeddings["description"]
+    )
+
+    community_reports_embeddings = final_community_reports[
+        ["id", "full_content", "summary", "title"]
+    ].copy()
+
+    if base_text_embed:
+        log.info("Creating embeddings")
+        # Embed raw_content in documents table
+        if document_raw_content_embedding in embedded_fields and isinstance(
+            documents_embeddings, pd.DataFrame
+        ):
+            await run_embeddings(
+                data=documents_embeddings,
+                column_to_embed="raw_content",
+                filename="create_final_documents_raw_content_embeddings",
+                callbacks=callbacks,
+                cache=cache,
+                storage=storage,
+                base_text_embed=base_text_embed,
+                message="Documents embeddings (raw content)",
+            )
+
+        # Embed description in relationships table
+        if relationship_description_embedding in embedded_fields and isinstance(
+            relationships_embeddings, pd.DataFrame
+        ):
+            await run_embeddings(
+                data=relationships_embeddings,
+                column_to_embed="description",
+                filename="create_final_relationships_description_embeddings",
+                callbacks=callbacks,
+                cache=cache,
+                storage=storage,
+                base_text_embed=base_text_embed,
+                message="Relationships embeddings (description)",
+            )
+
+        # Embed text in text_units table
+        if text_unit_text_embedding in embedded_fields and isinstance(
+            text_units_embeddings, pd.DataFrame
+        ):
+            await run_embeddings(
+                data=text_units_embeddings,
+                column_to_embed="text",
+                filename="create_final_text_units_text_embeddings",
+                callbacks=callbacks,
+                cache=cache,
+                storage=storage,
+                base_text_embed=base_text_embed,
+                message="Text units embeddings (text)",
+            )
+
+        # Embed name in entities table
+        if entity_name_embedding in embedded_fields and isinstance(
+            entities_embeddings, pd.DataFrame
+        ):
+            await run_embeddings(
+                data=entities_embeddings.copy(),
```

Comment:
This was already a copy, and now this calls copy again. II would even challenge the need to copy at all

Reply:
same answer than above